[#27] Implement DB serialization

Sheptytskyid opened the pull request.

KoliushkoArtem committed:
added database Serialization for HotelDao, UserDao

Sheptytskyid commented on src/main/java/com/manardenza/dao/HotelDaoImpl.java:
```diff
@@ -2,13 +2,18 @@
 
     private static HotelDaoImpl instance;
 
+    {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1171)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,63 @@
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(databaseFile))) {
+            database = (List<T>) ois.readObject();
+        } catch (IOException e) {
+            e.printStackTrace();
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,63 @@
+            database = (List<T>) ois.readObject();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ClassNotFoundException e) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Combine this catch with the one at line 31, which has the same body. (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2147)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,63 @@
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,63 @@
+
+    public void saveToFile() {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make the "java.util.List" class "Serializable" or don't write it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2118)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,63 @@
+    public void saveToFile() {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
+        } catch (IOException e) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,63 @@
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
+        } catch (IOException e) {
+            e.printStackTrace();
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,63 @@
+    public void readFromFile() {
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(databaseFile))) {
+            database = (List<T>) ois.readObject();
+        } catch (IOException e) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Sheptytskyid commented on src/main/java/com/manardenza/dao/UserDaoImpl.java:
```diff
@@ -2,12 +2,17 @@
 
     private static UserDaoImpl instance;
 
+    {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1171)

Sheptytskyid commented:
SonarQube analysis reported 10 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,63 @@
     protected List<T> database = new ArrayList<>();
+    protected File databaseFile;
+
+    public void saveToFile() {
```
Make this private

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,63 @@
+
+    }
+
+    public void readFromFile() {
```
Make this private

KoliushkoArtem committed:
implement HotelService all methods realization

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,61 @@
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(databaseFile))) {
+            database = (List<T>) ois.readObject();
+        } catch (IOException | ClassNotFoundException e) {
+            e.printStackTrace();
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

zozich replied:
You will need to do this - log this exception with the logger. As logging branch is not merged yet, use //FIXME: comments not to forget to fix this issue in future

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,61 @@
+
+    private void saveToFile() {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make the "java.util.List" class "Serializable" or don't write it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2118)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,61 @@
+    private void saveToFile() {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
+        } catch (IOException e) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

zozich replied:
//FIXME:

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,61 @@
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
+        } catch (IOException e) {
+            e.printStackTrace();
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

zozich replied:
//FIXME:

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,61 @@
+    private void readFromFile() {
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(databaseFile))) {
+            database = (List<T>) ois.readObject();
+        } catch (IOException | ClassNotFoundException e) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

zozich replied:
//FIXME:

Sheptytskyid commented:
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

zozich commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,61 @@
+    protected File databaseFile;
+
+    private void saveToFile() {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
```
Try storing ObjectOutputStream variable as an instance variable not to open new stream each time any database call is performed

zozich reviewed (changes requested):
Please amend

zozich commented on src/main/java/com/manardenza/entity/Hotel.java:
```diff
@@ -2,10 +2,11 @@
 
 @Data
-public class Hotel extends AbstractObject {
+public class Hotel extends AbstractObject implements Serializable {
```
If you have marked AbstractObject as Serializable, this class will become so as well

zozich commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -3,11 +3,12 @@
 @Data
 @AllArgsConstructor
-public class Reservation {
+public class Reservation implements Serializable {
```
If you have marked AbstractObject as Serializable, this class will become so as well

zozich commented on src/main/java/com/manardenza/entity/Room.java:
```diff
@@ -2,8 +2,10 @@
+
 @Data
-public class Room extends AbstractObject {
+public class Room extends AbstractObject implements Serializable{
```
If you have marked AbstractObject as Serializable, this class will become so as well

zozich commented on src/main/java/com/manardenza/entity/User.java:
```diff
@@ -2,8 +2,10 @@
+
 @Data
-public class User extends AbstractObject {
+public class User extends AbstractObject implements Serializable {
```
If you have marked AbstractObject as Serializable, this class will become so as well

zozich reviewed (changes requested):
Amend

KoliushkoArtem committed:
Fixed code based on comments

KoliushkoArtem committed:
Fixed conflicts
Merge branch 'master' of https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6 into 27_Implement_DB_serialization

# Conflicts:
#	src/main/java/com/manardenza/dao/HotelDaoImpl.java
#	src/main/java/com/manardenza/dao/UserDaoImpl.java

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+        try {
+            database = (List<T>) ois.readObject();
+            ois.close();
+        } catch (IOException | ClassNotFoundException ex) { //FIXME: Either log or rethrow this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+            database = (List<T>) ois.readObject();
+            ois.close();
+        } catch (IOException | ClassNotFoundException ex) { //FIXME: Either log or rethrow this exception.
+            ex.printStackTrace(); //FIXME: Use a logger to log this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+
+    private void saveToFile() {
+        try {
+            oos.writeObject(database);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make the "java.util.List" class "Serializable" or don't write it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2118)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+    protected FileInputStream fis = null;
+    protected ObjectInputStream ois = null;
+
+    {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1171)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+        try {
+            oos.writeObject(database);
+            oos.close();
+        } catch (IOException ex) { //FIXME: Either log or rethrow this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+            oos.writeObject(database);
+            oos.close();
+        } catch (IOException ex) { //FIXME: Either log or rethrow this exception.
+            ex.printStackTrace(); //FIXME: Use a logger to log this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+            fis = new FileInputStream(databaseFile);
+            ois = new ObjectInputStream(fis);
+
+        } catch (IOException ex) { //FIXME: Either log or rethrow this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+            ois = new ObjectInputStream(fis);
+
+        } catch (IOException ex) { //FIXME: Either log or rethrow this exception.
+            ex.printStackTrace(); //FIXME: Use a logger to log this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented:
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

KoliushkoArtem committed:
Fixed

Sheptytskyid commented on src/main/java/com/manardenza/dao/HotelDaoImpl.java:
```diff
@@ -1,14 +1,20 @@
 package com.manardenza.dao;
 
 import com.manardenza.entity.Hotel;
+import com.manardenza.utils.DatabaseSeeder;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.manardenza.utils.DatabaseSeeder'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

Sheptytskyid commented on src/main/java/com/manardenza/dao/HotelDaoImpl.java:
```diff
@@ -1,14 +1,20 @@
 
     private static HotelDaoImpl instance;
 
+    {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1171)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+    private void readFromFile() {
+        try {
+            database = (List<T>) ois.readObject();
+        } catch (IOException | ClassNotFoundException ex) { //FIXME: Either log or rethrow this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+        try {
+            database = (List<T>) ois.readObject();
+        } catch (IOException | ClassNotFoundException ex) { //FIXME: Either log or rethrow this exception.
+            ex.printStackTrace(); //FIXME: Use a logger to log this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+    private void saveToFile() {
+        try {
+            oos.writeObject(database);
+        } catch (IOException ex) { //FIXME: Either log or rethrow this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,78 @@
+        try {
+            oos.writeObject(database);
+        } catch (IOException ex) { //FIXME: Either log or rethrow this exception.
+            ex.printStackTrace(); //FIXME: Use a logger to log this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented on src/main/java/com/manardenza/dao/UserDaoImpl.java:
```diff
@@ -2,10 +2,17 @@
 import com.manardenza.entity.User;
 
+import java.io.File;
+import java.util.List;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'java.util.List'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

Sheptytskyid commented on src/main/java/com/manardenza/dao/UserDaoImpl.java:
```diff
@@ -2,10 +2,17 @@
 
     private static UserDaoImpl instance;
 
+    {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1171)

Sheptytskyid commented on src/main/java/com/manardenza/Main.java:
```diff
@@ -1,9 +1,10 @@
 package com.manardenza;
 
+import com.manardenza.utils.DatabaseSeeder;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.manardenza.utils.DatabaseSeeder'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

Sheptytskyid commented:
SonarQube analysis reported 18 issues
* ![MAJOR][MAJOR] 12 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

KoliushkoArtem committed:
Fixed

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,60 @@
+
+    private void saveToFile() {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make the "java.util.List" class "Serializable" or don't write it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2118)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,60 @@
+    private void saveToFile() {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
+        } catch (IOException e) { //FIXME: Either log or rethrow this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,60 @@
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
+        } catch (IOException e) { //FIXME: Either log or rethrow this exception.
+            e.printStackTrace();//FIXME: Use a logger to log this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,60 @@
+    private void readFromFile() {
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(databaseFile))) {
+            database = (List<T>) ois.readObject();
+        } catch (IOException | ClassNotFoundException e) { //FIXME: Either log or rethrow this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,60 @@
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(databaseFile))) {
+            database = (List<T>) ois.readObject();
+        } catch (IOException | ClassNotFoundException e) { //FIXME: Either log or rethrow this exception.
+            e.printStackTrace();//FIXME: Use a logger to log this exception.
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

Sheptytskyid commented:
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

KoliushkoArtem committed:
Fixed by Andrew comments

zozich reviewed (approved):
Cool!

Sheptytskyid commented on src/main/java/com/manardenza/dao/AbstractDao.java:
```diff
@@ -2,27 +2,64 @@
+
+    private void saveToFile() {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make the "java.util.List" class "Serializable" or don't write it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2118)
